### PR TITLE
Implements WebTemplate based validation

### PR DESCRIPTION
**Pull request by subigre**

### CODE REVIEW

File: serialisation/src/test/java/org/ehrbase/serialisation/flatencoding/FlatJsonTest.java [hunk 1]
```diff
@@ -402,12 +389,6 @@ private void check(
 
     softAssertions.assertThat(unmarshal).isNotNull();
 
-    try {
-      new Validator(templateProvider.find(templateId).get()).check(unmarshal);
-    } catch (Exception e) {
-      softAssertions.fail(e.getMessage());
-    }
```
Comment (stefanspiska) [inline serialisation/src/test/java/org/ehrbase/serialisation/flatencoding/FlatJsonTest.java:409]: Please add the new compostionValidate where ever you removed the old one.

File: serialisation/src/test/java/org/ehrbase/serialisation/flatencoding/FlatConformanceTest.java [hunk 2]
```diff
@@ -297,12 +296,6 @@ private void check(
 
     softAssertions.assertThat(unmarshal).isNotNull();
 
-    try {
-      new Validator(templateProvider.find(templateId).get()).check(unmarshal);
-    } catch (Exception e) {
-      softAssertions.fail(e.getMessage());
-    }
-
```
Comment (stefanspiska) [inline serialisation/src/test/java/org/ehrbase/serialisation/flatencoding/FlatConformanceTest.java:305]: Please add the new compostionValidate where ever you removed the old one.

File: serialisation/src/test/java/org/ehrbase/serialisation/dbencoding/DBEncodeTestWithValidation.java [hunk 3]
```diff
@@ -1,81 +0,0 @@
-/*
- * Copyright (c) 2019 Vitasystems GmbH and Hannover Medical School.
```
Comment (stefanspiska) [inline serialisation/src/test/java/org/ehrbase/serialisation/dbencoding/DBEncodeTestWithValidation.java:2]: why was this deleted

File: serialisation/src/test/java/org/ehrbase/serialisation/dbencoding/DBEncodeTest.java [hunk 4]
```diff
@@ -702,21 +698,6 @@ public void testEncodeDvTextAsJson(){
         assertThat(converted.getAsJsonObject().get("_type").getAsString()).isEqualTo("DV_CODED_TEXT");
     }
 
-    @Test
```
Comment (stefanspiska) [inline serialisation/src/test/java/org/ehrbase/serialisation/dbencoding/DBEncodeTest.java:705]: why was this deleted?

File: serialisation/src/test/java/org/ehrbase/serialisation/flatencoding/StructuredJsonTest.java [hunk 5]
```diff
@@ -21,72 +21,62 @@
...
 
-        softAssertions.assertThat(unmarshal).isNotNull();
+    SoftAssertions softAssertions = new SoftAssertions();
 
-        try {
-            new Validator(templateProvider.find(templateId).get()).check(unmarshal);
-        } catch (Exception e) {
-            softAssertions.fail(e.getMessage());
-        }
+    softAssertions.assertThat(unmarshal).isNotNull();
```
Comment (stefanspiska) [inline serialisation/src/test/java/org/ehrbase/serialisation/flatencoding/StructuredJsonTest.java:67]: Please add the new compostionValidate where ever you removed the old one.

File: serialisation/src/test/java/org/ehrbase/serialisation/flatencoding/std/umarshal/FlatJsonUnmarshallerTest.java [hunk 6]
```diff
@@ -64,12 +64,6 @@ public void unmarshal() throws IOException, XmlException {
         "2020-05-11T22:53:12.039139+02:00");
     assertThat(observation.getSubject()).isNotNull();
     assertThat(observation.getSubject().getClass()).isEqualTo(PartySelf.class);
-
```
Comment (stefanspiska) [inline serialisation/src/test/java/org/ehrbase/serialisation/flatencoding/std/umarshal/FlatJsonUnmarshallerTest.java:67]: Please add the new compostionValidate where ever you removed the old one.

**Review by stefanspiska — CHANGES_REQUESTED** [review]

File: serialisation/src/test/java/org/ehrbase/serialisation/flatencoding/FlatJsonTest.java (hunk 1 above)
Comment (subigre) [reply serialisation/src/test/java/org/ehrbase/serialisation/flatencoding/FlatJsonTest.java:409]: I cannot use CompositionValidator here because validation module now depends on serialization and we will have a circular dependency issue.
image

But to keep these test cases, I have created the [FlatCompositionValidatorTest](https://github.com/ehrbase/openEHR_SDK/blob/CDR-45_web_template_validation/validation/src/test/java/org/ehrbase/validation/FlatCompositionValidatorTest.java) that validates the same compositions.

File: serialisation/src/test/java/org/ehrbase/serialisation/flatencoding/StructuredJsonTest.java (hunk 5 above)
Comment (subigre) [reply serialisation/src/test/java/org/ehrbase/serialisation/flatencoding/StructuredJsonTest.java:67]: Tests added in `StructuredJsonValidationTest`

File: serialisation/src/test/java/org/ehrbase/serialisation/flatencoding/FlatConformanceTest.java (hunk 2 above)
Comment (subigre) [reply serialisation/src/test/java/org/ehrbase/serialisation/flatencoding/FlatConformanceTest.java:305]: Tests added in `FlatConformanceValidationTest`

File: serialisation/src/test/java/org/ehrbase/serialisation/flatencoding/std/umarshal/FlatJsonUnmarshallerTest.java (hunk 6 above)
Comment (subigre) [reply serialisation/src/test/java/org/ehrbase/serialisation/flatencoding/std/umarshal/FlatJsonUnmarshallerTest.java:67]: The validation of these FLAT compositions is tested in `FlatValidationTest`

**Review by stefanspiska — APPROVED** [review]
